### PR TITLE
Fix stdarg and make "make test" actually work.

### PR DESCRIPTION
**Pull request by vathpela**
Fix stdarg.h / va_list to all be the same everywhere, and move more of our standard library implementations all to the same spot.

This goes on top of https://github.com/rhboot/shim/pull/308

### CODE REVIEW

File: Cryptlib/Include/OpenSslSupport.h [hunk 1]
```diff
@@ -15,6 +15,30 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
...
+#endif
+
+/*
+ * Include stddef.h to avoid redefining "offsetof"
+ */
+#include <stddef.h>
+#include <stdarg.h>
+#include <stdint.h>
+#include <string.h>
+
```
Comment (jsetje) [inline Cryptlib/Include/OpenSslSupport.h:41]: Nit: Carriage Returns

File: Cryptlib/Include/OpenSslSupport.h [hunk 2]
```diff
@@ -328,7 +227,7 @@ size_t         fwrite      (const void *, size_t, size_t, FILE *);
 char           *fgets      (char *, int, FILE *);
 int            fputs       (const char *, FILE *);
 int            fprintf     (FILE *, const char *, ...);
-int            vfprintf    (FILE *, const char *, VA_LIST);
+int            vfprintf    (FILE *, const char *, va_list);
```
Comment (jsetje) [inline Cryptlib/Include/OpenSslSupport.h:230]: nit: CR

File: Cryptlib/Include/OpenSslSupport.h [hunk 3]
```diff
@@ -358,7 +256,7 @@ extern FILE  *stderr;
 extern FILE  *stdin;
 extern FILE  *stdout;
 
-#define AsciiStrLen(x) strlena(x)
+#define AsciiStrLen(x) strlen(x)
```
Comment (jsetje) [inline Cryptlib/Include/OpenSslSupport.h:259]: nit: CR

File: Cryptlib/InternalCryptLib.h [hunk 4]
```diff
@@ -15,6 +15,8 @@ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
 #ifndef __INTERNAL_CRYPT_LIB_H__
 #define __INTERNAL_CRYPT_LIB_H__
 
+#include <stdarg.h>
+
```
Comment (jsetje) [inline Cryptlib/InternalCryptLib.h:19]: nit: CR

File: Cryptlib/Include/OpenSslSupport.h (hunk 1 above)
Comment (vathpela) [reply Cryptlib/Include/OpenSslSupport.h:41]: What about them?  This leaves the existing file format intact.

**Pull request merged**